dace/transformation/dataflow/map_distribution.py

```diff
@@ -0,0 +1,940 @@
+    mid_array = pm.PatternNode(nodes.AccessNode)
+    scatter = pm.PatternNode(nodes.Tasklet)
+    out_array = pm.PatternNode(nodes.AccessNode)
+
+    @staticmethod
+    def expressions():
+        return [
+            sdutil.node_path_graph(RedundantComm2D.in_array,
+                                   RedundantComm2D.gather,
+                                   RedundantComm2D.mid_array,
+                                   RedundantComm2D.scatter,
+                                   RedundantComm2D.out_array)
+        ]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        gather = graph.nodes()[candidate[RedundantComm2D.gather]]
+        if '_block_sizes' not in gather.in_connectors:
+            return False
+        scatter = graph.nodes()[candidate[RedundantComm2D.scatter]]
+        if '_gdescriptor' not in scatter.out_connectors:
+            return False
+        in_array = graph.nodes()[candidate[RedundantComm2D.in_array]]
+        out_array = graph.nodes()[candidate[RedundantComm2D.out_array]]
+        in_desc = in_array.desc(sdfg)
+        out_desc = out_array.desc(sdfg)
+        if len(in_desc.shape) != 2:
+            return False
+        if in_desc.shape == out_desc.shape:
+            return True
+        return False
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        in_array = graph.nodes()[candidate[RedundantComm2D.in_array]]
+
+        return "Remove " + str(in_array)
+
+    def apply(self, sdfg):
+        graph = sdfg.nodes()[self.state_id]
+        in_array = self.in_array(sdfg)
+        gather = self.gather(sdfg)
+        mid_array = self.mid_array(sdfg)
+        scatter = self.scatter(sdfg)
+        out_array = self.out_array(sdfg)
+
+        in_desc = sdfg.arrays[in_array.data]
+        out_desc = sdfg.arrays[out_array.data]
+
+        for e in graph.in_edges(gather):
+            if e.src != in_array:
+                if graph.in_degree(e.src) == 0 and graph.out_degree(e.src) == 1:
+                    graph.remove_edge(e)
+                    graph.remove_node(e.src)
+        for e in graph.out_edges(scatter):
+            if e.dst != out_array:
+                if graph.in_degree(e.dst) == 1 and graph.out_degree(e.dst) == 0:
+                    graph.remove_edge(e)
+                    graph.remove_node(e.dst)
+        
+        for e in graph.out_edges(out_array):
+            path = graph.memlet_tree(e)
+            for e2 in path:
+                if e2.data.data == out_array.data:
+                    e2.data.data = in_array.data
+            graph.remove_edge(e)
+            graph.add_edge(in_array, None, e.dst, e.dst_conn,
+                           dace.Memlet.from_array(in_array, in_desc))
+
+        graph.remove_node(gather)
+        graph.remove_node(mid_array)
+        graph.remove_node(scatter)
+        graph.remove_node(out_array)
+
+
+@registry.autoregister_params(singlestate=True)
+class StencilOperation(pm.Transformation):
```

Comment:
Do we use this transformation?

Reply:
No, it is part of the stuff that we left partially implemented for SC, i.e., if it goes now in another PR it will die.